### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by sondrelg**
Hi, 

I was about to suggest a few improvements to the docker-compose setup, but saw a few minor things that might be improved that I wanted to create a dedicated PR for. 

Feel free to disagree with the changes made. I'll try to add comments explaining each of them.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-default_language_version:
```
Comment (sondrelg) [inline .pre-commit-config.yaml:1]: This crashed my pre-commit setup when running on 3.9. See commit message for details.

File: mypy.ini [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [mypy]
 plugins = pydantic.mypy, sqlalchemy.ext.mypy.plugin
-
+python_version = 3.10
```
Comment (sondrelg) [inline mypy.ini:3]: Again, running this in a 3.9 venv caused mypy to infer the wrong python version. Better to be specific.

File: pyproject.toml [hunk 3]
```diff
@@ -29,6 +29,7 @@ line-length = 120
 include = '\.pyi?$'
 
 [tool.isort]
+line_length = 120
```
Comment (sondrelg) [inline pyproject.toml:32]: To match other linters and checkers

File: pyproject.toml [hunk 4]
```diff
@@ -61,4 +62,14 @@ ignored-argument-names = "args|kwargs|_|__"
 good-names = "_,i,e,v,T"
 
 [tool.coverage.run]
-omit = ["*/tests/*"]
```
Comment (sondrelg) [inline pyproject.toml:64]: Feel free to disagree with this. The maintainer of coverage recommends *including* tests in coverage, as it's a good way to detect dead tests and unused test utilities.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (yudjinn) [reply .pre-commit-config.yaml:1]: this was mostly done by @Goldziher  so I'm unsure of the decisions behind the params here.

**Review by yudjinn — APPROVED** [review]

File: mypy.ini (hunk 2 above)
Comment (Goldziher) [reply mypy.ini:3]: But the code uses 3.10 style unions, its not 3.9 comp

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -24,7 +22,7 @@ repos:
     rev: 21.12b0
     hooks:
       - id: black
-        args: [ --config=./pyproject.toml ]
+        args: [ "--quiet" ]
```
[the remark this answers is not in the document]
Comment (Goldziher) [reply .pre-commit-config.yaml:25]: well, the point is to let users adjust the template, no?

**Pull request merged**